tests/e2e/test_linker_run.py

```diff
@@ -0,0 +1,73 @@
+import subprocess
+import sys
+import tempfile
+from pathlib import Path
+
+import pytest
+
+from linker.utilities.data_utils import load_yaml
+from linker.utilities.slurm_utils import is_on_slurm
+
+SPECIFICATIONS_DIR = Path("tests/e2e/specifications")
+
+
+@pytest.mark.slow
+@pytest.mark.skipif(
+    not is_on_slurm(),
+    reason="Must be on slurm to run this test.",
+)
+@pytest.mark.parametrize(
+    "pipeline_specification, input_data, computing_environment",
+    [
+        # local
+        (
+            "pipeline.yaml",
+            "input_data.yaml",
+            "environment_local.yaml",
+        ),
+        # slurm
+        (
+            "pipeline.yaml",
+            "input_data.yaml",
+            "environment_slurm.yaml",
+        ),
+    ],
+)
+def test_linker_run(pipeline_specification, input_data, computing_environment, capsys):
+    """e2e tests for 'linker run' command"""
+    # Create a temporary directory to store results. We cannot use pytest's tmp_path fixture
+    # because other nodes do not have access to it.
+    with tempfile.TemporaryDirectory(dir="tests/e2e/") as results_dir:
+        results_dir = Path(results_dir)
+        cmd = (
+            "linker run "
+            f"-p {SPECIFICATIONS_DIR / pipeline_specification} "
+            f"-i {SPECIFICATIONS_DIR / input_data} "
+            f"-e {SPECIFICATIONS_DIR / computing_environment} "
+            f"-o {str(results_dir)} "
+            "--no-timestamp"
+        )
+        with capsys.disabled():  # disabled so we can monitor job submissions
+            print(
+                "\n\n*** RUNNING TEST ***\n"
+                f"[{pipeline_specification}, {input_data}, {computing_environment}]\n"
+            )
+            subprocess.run(
```

Comment:
I've never worked with this before, but would Click's built-in testing functions work here instead of doing subprocess?
https://click.palletsprojects.com/en/8.1.x/testing/

Reply:
Hmm, maybe? I'll check it out - I didn't know this existed

Reply:
This is proving to be tricker than I want. Based on quickly giving it a go, it seems like you can't `runner.invoke` in a parameterized test. I'm going to leave it as is for now unless you think there's good reason to move away from subprocess.